pymc_experimental/model_transform/conditioning.py

```diff
@@ -0,0 +1,103 @@
+from typing import Any, Dict, List, Sequence, Union
+
+from pymc import Model
+from pymc.pytensorf import _replace_rvs_in_graphs
+from pytensor.tensor import TensorVariable
+
+from pymc_experimental.utils.model_fgraph import (
+    ModelFreeRV,
+    extract_dims,
+    fgraph_from_model,
+    model_from_fgraph,
+    model_named,
+    model_observed_rv,
+    toposort_replace,
+)
+
+
+def observe(model: Model, vars_to_observations: Dict[Union["str"], Any]) -> Model:
+    """Convert free RVS to observed RVs.
+
+    This function returns a new model.
+    """
+    vars_to_observations = {
+        model[var] if isinstance(var, str) else var: obs
+        for var, obs in vars_to_observations.items()
+    }
+
+    # Note: Since PyMC can infer logprob expressions we could also allow observing Deterministics
```

Comment:
Since we already are in experimental, do you want to actually try this out?

Reply:
Sure :D